aiidalab_widgets_base/nodes.py

```diff
@@ -278,8 +278,12 @@ def _update_tree_node(self, tree_node):
         if isinstance(tree_node, AiidaProcessNodeTreeNode):
             process_node = load_node(tree_node.pk)
             tree_node.name = calc_info(process_node)
-            tree_node.icon_style = self.PROCESS_STATE_STYLE.get(
-                process_node.process_state, self.PROCESS_STATE_STYLE_DEFAULT
+            tree_node.icon_style = (
+                self.PROCESS_STATE_STYLE.get(
+                    process_node.process_state, self.PROCESS_STATE_STYLE_DEFAULT
+                )
+                if not process_node.is_failed
+                else "danger"
             )
```

Comment:
So the one thing that bugs me is the hard-coded `"danger"` and I think the easiest way to avoid this is to simply determine the state first:

```suggestion
            # Override the process state in case that the process node has failed:
            # (This could be refactored with structural pattern matching with py>=3.10.)
            process_state = ProcessState.EXCEPTED if process_node.is_failed else process_node.process_state
            tree_node.icon_style = self.PROCESS_STATE_STYLE.get(process_state, PROCESS_STATE_STYLE_DEFAULT)
```
_The line breaks are probably wrong, but the pre-commit hooks should fix it._